app/helpers/application_helper.rb

```diff
@@ -1,2 +1,109 @@
+require 'net/http'
+require 'uri'
+
 module ApplicationHelper
-end
+
+	def shift_notification(shift, title: nil, content: nil, test: false, min_before: 10, send_now: false)
+		# Send to all shift members
+		netids = shift.users.collect(&:netid)
+		# Notification time is min_before minutes before shift, if not sending now
+		time = send_now ? nil : shift.start_time - min_before*60
+		title = "Upcoming Shift Reminder" if title.nil?
+		if content.nil?
+			content = "Hey! Your tent shift in K-Ville starts in #{min_before} minutes!"
+		end
```

Comment:
I like that you kept the message dynamic with {min_before}, later down the road users can possibly adjust when they receive their notification.